doc/api/stream.md

```diff
@@ -1158,8 +1160,9 @@ myReader.on('readable', () => {
 added: v8.0.0
 -->
 
-Destroy the stream, and emit `'error'`. After this call, the
-readable stream will release any internal resources.
+Destroy the stream, and emit `'error'` and `close`. After this call, the
+readable stream will release any internal resources and subsequent calls
+to `read` will result in an `ERR_STREAM_DESTROYED` error.
```

Comment:
I have code that rely on the current behavior (`read` after destroy) to ensure that all buffered data is written out.

```js
socket.pipe(parser);
socket.on('close', () => {
  // Ensure that all buffered data is written to the parser if the socket is
  // destroyed due to a network error (ECONNRESET, EPIPE, etc.) and the `'end'`
  // event is not emitted.
  socket.read();
  parser.end();
});
```

I guess that with this change it will no longer work, how can I fix that?

Reply:
This is actually a typo. It should be '.push' and now `.read'. Fixed it just now. Also you are right that you should be able to get data out of the buffer after .destroy(). That is also fixed.

Reply:
Awesome, thanks.